File: Ruby-Labyrinth/weapon.rb
# frozen_string_literal: true
#encoding : UTF−8
class Weapon
  def initialize(power, uses)
    @power = power
    @uses = uses
  end
  def attack
    if @uses > 0
      @uses -= 1
      return @power
    else
      return 0
    end
  end

  def to_s
    "W[" + @power.to_s + "," + @uses.to_s + "]"
  end
  def discard
    d = Dice.new
    d.discardElement(@uses)
  end
end

File: Java-Labyrinth/src/coffee/labyrinth/Weapon.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package coffee.labyrinth;

/**
 *
 * @author albertoct
 */
public class Weapon {
    private float power;
    private int uses;
    
    public Weapon(float power,int uses){
        this.power=power;
        this.uses=uses;
    }
    
    public float attack(){
        if(uses>0){
            return power;
        }else{
            return 0;
        }
    }
    
    public String toString(){
        return "W[" + power + "," + uses + "]";
    }
    
    public boolean discard(){
        Dice d =new Dice();
        return d.discardElement(uses);
    }
}
